libdevice/fallback-cassert.cpp

```diff
@@ -27,6 +53,62 @@ DEVICE_EXTERN_C void __devicelib_assert_fail(const char *expr, const char *file,
                      // (func) ? func : "<unknown function>",
                      func, gid0, gid1, gid2, lid0, lid1, lid2, expr);
 
+  {
+    int Expected = ASSERT_NONE;
+    int Desired = ASSERT_START;
+
+    if (CompareAndSet(&__SYCL_AssertHappenedMem.Flag, Desired, Expected) ==
+        Expected) {
+      __SYCL_AssertHappenedMem.Line = line;
+      __SYCL_AssertHappenedMem.GID0 = gid0;
+      __SYCL_AssertHappenedMem.GID1 = gid1;
+      __SYCL_AssertHappenedMem.GID2 = gid2;
+      __SYCL_AssertHappenedMem.LID0 = lid0;
+      __SYCL_AssertHappenedMem.LID1 = lid1;
+      __SYCL_AssertHappenedMem.LID2 = lid2;
+
+      int ExprLength = 0;
+      int FileLength = 0;
+      int FuncLength = 0;
+
+      if (expr)
+        for (const char *C = expr; *C != '\0'; ++C, ++ExprLength)
+          ;
```

Comment:
Do you plan to add more code here?  Otherwise, you could have used `std::strlen`.

Reply:
Is it available in libdevice i.e. in device code?
@jinge90 , could you, please, confirm it?

Reply:
I don't think it's worth it to introduce another dependency here with `std::strlen`.